code/datums/components/virtual_entity.dm

```diff
@@ -5,7 +5,14 @@
 
 /datum/component/virtual_entity/Initialize()
 	. = ..()
+
+	for(var/obj/machinery/quantum_server/server in SSmachines.get_machines_by_type(/obj/machinery/quantum_server))
+		if(server.obj_flags & EMAGGED)
+			jailbreak_mobs() //This just sends a message and self-deletes, a bit messy but it works.
+			return
+
 	RegisterSignal(parent, COMSIG_MOVABLE_PRE_MOVE, PROC_REF(on_parent_pre_move))
+	RegisterSignal(parent, COMSIG_BITRUNNER_SERVER_EMAGGED, PROC_REF(jailbreak_mobs))
```

Comment:
does parent receive this signal or ssdcs? if we had a ref to the server we could conditionally register it if it wasn't already emagged

Reply:
It gets sent as a global signal. I'm assuming that means ssdcs get sent this signal here.

Reply:
have you tested it

Reply:
okay you were right on this, parent was DCS idk what I was thinking